.github/workflows/nightly-ci.yml

```diff
@@ -53,13 +53,13 @@ jobs:
         # on that approach. So I'm just disabling the particular lint globally.
         run: cargo clippy --all-targets --all-features -- --allow non_local_definitions -Dwarnings
       - name: Build
-        run: cargo build --all-targets --all-features
+        run: cargo build --workspace --all-targets --all-features
```

Comment:
I discovered (by running it) that this will not run all of the test binaries if a test fails in one of the crates.

We should consider using the [`--no-fail-fast` flag](https://doc.rust-lang.org/cargo/commands/cargo-test.html) as well:
> Run all tests regardless of failure. Without this flag, Cargo will exit after the first executable fails. The Rust test harness will run all tests within the executable to completion, this flag only applies to the executable as a whole.

Based on what I just ran in my checkout, this flag still has the process exit with non-zero if any of the tests failed - we might be able to just drop it in here?

Reply:
Oh, that's nice, it even lists the targets which failed:
```
error: 2 targets failed:
    `-p cells --doc`
    `-p tiledb-common --doc`
```
(for example)